### ISSUE TITLE
Component handles

### ISSUE THREAD

**Issue by ayebear** [Lastresort92/OCS#13]
Add support for safe component "handles" instead of using raw pointers to vector elements which could be invalidated. They could contain a pointer/reference to the PackedArray instance, and an ID to the element in the internal vector.

A possibly more generic solution would be to add a smart handle to PackedArray, which would contain "this" and the ID. But it would work like a normal pointer, with the \* and -> operators.

Basically, just try to solve this problem:

```
auto obj1 = objects.createObject();
auto obj2 = objects.createObject();

objects.addComponents<Position>(obj1);
auto pos1 = object.getComponent<Position>(obj1);

objects.addComponents<Position>(obj2);
auto pos2 = object.getComponent<Position>(obj2);

pos1->x = 0; // This *could* fail, if the component array resized after adding the second component
```
